### PR TITLE
Provide nouislider as npm dependency

### PR DESCRIPTION
**Pull request by bobisjan**

### CODE REVIEW

**Comment by kennethkalmer** [comment]
Thanks for the contribution! I'll spend some time playing with it and understanding the changes.

First question though, why does the shim's location change? I understand the shim itself changes to accommodate the NPM package, but why the path too?

**Comment by bobisjan** [comment]
Happy to contribute 😊

> First question though, why does the shim's location change? I understand the shim itself changes to accommodate the NPM package, but why the path too?

Just to align with Ember CLI - this is a [default path](https://github.com/ember-cli/ember-cli/tree/master/blueprints/vendor-shim/files/vendor/shims) generated by `ember generate vendor-shim`.

<hr>

FYI I was following [this article](https://simplabs.com/blog/2017/02/13/npm-libs-in-ember-cli.html) when moving `nouislider` to `npm`.

**Comment by rwwagner90** [comment]
Just FYI, you can use this instead of all the tree stuff and custom broccoli work https://github.com/dfreeman/ember-cli-node-assets

**Comment by rwwagner90** [comment]
Would love to see this moved to npm though!

**Comment by rwwagner90** [comment]
If you guys would like me to PR in ember-cli-node-assets, let me know. Otherwise, this PR should be fine too. Need this in npm and no depending on ember in bower before we can update our app's ember version 😄

**Comment by kennethkalmer** [comment]
Going through this and all the other PR's now :)

**Comment by kennethkalmer** [comment]
Lazy question to you all, what is the minimum Ember version supported with this setup?

File: index.js [hunk 1]
```diff
@@ -1,26 +1,45 @@
...
-        production: bowerDirectory + '/nouislider/distribute/nouislider.min.js'
-      });
-      importShim.import(bowerDirectory + '/nouislider/distribute/nouislider.min.css');
+    return new MergeTrees([vendorTree, nouisliderTree]);
+  },
 
-      importShim.import('vendor/nouislider/shim.js', {
-        exports: { 'noUiSlider': ['default'] }
-      });
+  _ensureThisImport: function() {
```
Comment (kennethkalmer) [inline index.js]: OK, so this shim makes us support versions before 2.7... Wonder how far back it goes?

**Comment by kennethkalmer** [comment]
Looks solid afaict, thanks so much!

**Comment by kennethkalmer** [comment]
v0.12.0 has just been released, thanks for your contributions!

**Comment by bobisjan** [comment]
Based on the [ember-getowner-polyfill](https://github.com/rwjblue/ember-getowner-polyfill#compatibility), where `import` polyfill is used and tested, it should work well with Ember 1.10.

**Pull request merged**